app/main.py

```diff
@@ -1 +1,49 @@
-# write your code here
+class SoftwareEngineer:
+    skills = []
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def learn_skill(self, skill: str) -> None:
+        self.skills.append(skill)
+
+
+class FrontendDeveloper(SoftwareEngineer):
+    def __init__(self, name: str) -> None:
+        super().__init__(name)
+        if self.__class__ == FullStackDeveloper:
+            self.skills += ["JavaScript", "HTML", "CSS"]
+        else:
+            self.skills = ["JavaScript", "HTML", "CSS"]
```

Comment:
It is not needed. Just extend the `self.skills` list from parent class in 1 line

Reply:
Actually, I don`t see how...

Reply:
Done